x/ccv/provider/keeper/grpc_query.go

```diff
@@ -71,3 +72,55 @@ func (k Keeper) QueryConsumerChainStops(goCtx context.Context, req *types.QueryC
 
 	return &types.QueryConsumerChainStopProposalsResponse{Proposals: &props}, nil
 }
+
+func (k Keeper) QueryValidatorConsumerAddr(goCtx context.Context, req *types.QueryValidatorConsumerAddrRequest) (*types.QueryValidatorConsumerAddrResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	if _, found := k.GetConsumerClientId(ctx, req.ChainId); !found {
+		return nil, types.ErrUnknownConsumerChainId
+	}
```

Comment:
As discussed on call, this check should be removed because keys should be assignable/queryable before chain addition.